User-Help/AskingGuidance.md

```diff
@@ -0,0 +1,37 @@
+
+**Do some research**
+
+Before asking a new question, first take a look around. Has your question been asked before here on Codidact? You can [do a search on Codidact](/help/search) for keywords related to your question, or take to your favorite search engine to see if your answer is already there on Codidact.
+
+In addition to just making sure your question hasn't been asked already here on Codidact, take a few moments to search beyond the site. If you put your question title into a search engine, can you find the answer to your question in the first three results? If so, perhaps consider [alternative ways of sharing that information](/help/articles-blogs) here on Codidact, or writing a self-answered question to share that knowlege.
+
+**Include your end goal**
+
+It's always a good idea to include what exactly you're trying to achieve when you ask a question. It helps to provide context to your question, and can influence answers in ways you possibly hadn't considered.
+
+**Avoid too much subjectivity**
+
+Questions like "What's your favorite way to eat an egg?" can't be answered objectively - the answer, naturally, changes depending on who you ask. On the other hand, "What's the difference between hard-boiled eggs and an omelette?" can be answered without relying too much on personal opinions. Try to ask questions that can be answered with some level of objectivity.
+
+**Be specific**
+
+Asking really broad questions often isn't very useful. Instead, narrow your question down to ask about one thing in particular. Asking "How do I care for a dragon?" would need several full-size books to answer, but "Is garlic safe for my dragon to eat?" is much more manageable. Be specific when asking a question, and don't ask questions that would need answers the size of books.
+
+**Include details**
+
+The more details you include in your question, the better. Include any details that are relevant to your question, no matter how small - you never know what might be relevant!
+
+**Include examples and sources**
+
+When you're asking a question, it's almost always a good idea to include your *sources* and/or examples, depending on what exactly your question is about. You can almost always find a source to include that will provide some background information on whatever you're asking about; and including an example to show us what you're talking about, whether that's a photo, a writing sample, code fragment, or anything else, helps to improve your question.
+
+If you want to make sure that your question is a good fit for this site, you can always head over to our [Codidact Communities Discord server](https://discord.gg/jwjUA26), find the channel for this site, and ask if anybody would be willing to look over your question before you ask it.
```

Comment:
Suggested addition at the end:

**Writing the question can help you understand the question**

Sometimes the act of writing the question helps you notice important details and can even change what you end up asking in the end.  You start out thinking you have a question about omelettes and, in writing it, realize that your question is really about skillets.  This is why, when asking a question, we ask you to write the question first and *then* give it a title. By the time you've explained your problem you should have discovered a clear, succinct title, which is the first thing people will see.  Good questions have good titles.